nilearn/glm/first_level/first_level.py

```diff
@@ -1029,6 +1029,13 @@ def _prepare_mask(self, run_img):
         if isinstance(run_img, SurfaceImage) and not isinstance(
             self.mask_img, SurfaceMasker
         ):
+            if self.smoothing_fwhm is not None:
+                warn(
+                    "Parameter smoothing_fwhm is not "
+                    "supported for surface data",
+                    UserWarning,
+                    stacklevel=2,
```

Comment:
```suggestion
                    stacklevel=3,
```

I think that if we want the warning to reference the `fit` method (the first public method in the stack) this should be 3.